cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -152,8 +269,24 @@ impl ClientTrait for MockClient {
     async fn resolve(
         &self,
         _package_groups: Vec<PackageGroup>,
-    ) -> Result<Vec<ResolvedPackageGroup>, ResolveError> {
-        unimplemented!()
+    ) -> Result<ResolvedGroups, ResolveError> {
+        let mock_resp = self
+            .mock_responses
+            .lock()
+            .expect("couldn't acquire mock lock")
+            .pop_front();
+        if let Some(Response::Search(_)) = mock_resp {
+            panic!("found search response, expected resolve response");
+        } else if mock_resp.is_none() {
+            panic!("expected mock response, found nothing");
+        } else if let Some(Response::Error(err)) = mock_resp {
+            return Err(ResolveError::Resolve(APIError::ErrorResponse(
+                err.try_into()?,
+            )));
+        } else if let Some(Response::Resolve(resp)) = mock_resp {
+            return Ok(resp);
+        }
+        return Err(MockDataError::InvalidData("unrecognized response".into()).into());
```

Comment:
suggestion (nb): use a `match` instead

```suggestion
        match mock_resp {
            None => panic!("expected mock response, found nothing"),
            Some(Response::Search(_)) = {
                panic!("found search response, expected resolve response");
            },
            Some(Response::Error(err)) {
                return Err(ResolveError::Resolve(APIError::ErrorResponse(
                    err.try_into()?,
                )));
            }.
            Some(Response::Resolve(resp)) => return Ok(resp),
            _ => {
                return Err(MockDataError::InvalidData("unrecognized response".into()).into());
            }
        }
```

Reply:
Fixed